### PR TITLE
feat: add pipelines for testing in gitlab

### PR DESCRIPTION
**Pull request by 21adrian1996**
# Checklist (Only for frontend changes)
- [ ] Changes tested in IE11
- [ ] Responsive
- [ ] Documented "How to use it"
- [ ] Documented "How to install"
- [ ] Featured in Styleguide

# Summary of this PR

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - pip install -r requirements/test.txt
+    - flake8
+    - pytest -c pytest-ci.ini
+  tags:
+    - docker
+
+# this is for testing the frontend
+test-frontend:
+  stage: test
+  variables:
```
Comment (JonasNiestroj) [inline .gitlab-ci.yml:25]: We dont need the postgres password for the frontend tests

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+test-backend:
+  stage: test
+  variables:
+    POSTGRES_PASSWORD: unsafe
+  services:
+    - postgres:latest
+  script:
+    - cd backend
+    - pip install -r requirements/test.txt
+    - flake8
```
Comment (JonasNiestroj) [inline .gitlab-ci.yml:17]: Maybe split linting and testing?

Comment (schtibe) [reply .gitlab-ci.yml:17]: I can also be integrated into pytest: https://github.com/tholo/pytest-flake8
Don't know whether there's an advante...

Comment (JonasNiestroj) [reply .gitlab-ci.yml:17]: Yeah but i think then we should rename it to test-and-lint or similiar. For me test is only unit testing

**Pull request merged**